### DISCUSSION TITLE
Does 'shard_map' jit-compile the decorated function?

### DISCUSSION THREAD

**Discussion by acec2127** [jax-ml/jax#24161] [Q&A] [answered]
Really simple question, but I cannot find any answer to this question from the Jax documentation. If I decorate a function with 'shard_map', will it also be jitted, or do I need to decorate it with 'jit'? In which case, what is the most efficient order of composition order? Thanks!

**Comment by yashk2810** [comment, marked as answer]
No, shard_map is eager by default. You will need to wrap a jit around it. `jit(shard_map)` should be fine for your last question.

**Reply by acec2127** [reply]
Yes that's what I thought. Thanks for the answer!
